velox/functions/sparksql/tests/StringTest.cpp

```diff
@@ -995,5 +996,298 @@ TEST_F(StringTest, trim) {
       trimWithTrimStr("\u6570", "\u6574\u6570 \u6570\u636E!"),
       "\u6574\u6570 \u6570\u636E!");
 }
+
+TEST_F(StringTest, split) {
+  auto limit = -1;
+  auto delim = ",";
+  auto numRows = 3;
+  auto data = std::vector<std::string>{
+      {"I,he,she,they"}, // Simple
+      {"one,,,four,"}, // Empty strings
+      {""}, // The whole string is empty
+  };
+
+  auto createStrings = [&](RowSet /*rows*/) {
+    return makeFlatVector<StringView>(
+        numRows, [&](vector_size_t row) { return StringView{data[row]}; });
+  };
+
+  auto createDelims = [&](RowSet /*rows*/) {
+    return makeFlatVector<StringView>(
+        numRows, [&](vector_size_t row) { return StringView{delim}; });
+  };
+
+  auto createLimits = [&](RowSet /*rows*/) {
+    return makeFlatVector<int32_t>(
+        numRows, [&](vector_size_t row) { return limit; });
+  };
+
+  // No limit.
+  auto strings = createStrings({});
```

Comment:
Add a new Helper function in class StringTest
`testSplit(const VectorPtr& expected, const VectorPtr& input, std::optional<>limit)` will help code clean

Reply:
And function testSplit(const VectorPtr& expected, const VectorPtr& input)